asllib/ASLTypingReference.mld

```diff
@@ -1041,30 +1078,92 @@ Formally, one of the following applies:
   {3 TypingRule.PGeq}
 
     {4 Rule}
+    All of the following applies:
+    - [p] is the pattern which matches anything greater than or equal to an expression [e];
+    - [t_e, e'] is the result of annotating [e] in [env]; 
+    - [e'] is pure;
+    - [t] has the structure of an integer;
+    - [t_e] has the structure of an integer;
+    - [new_p] is the pattern which matches anything greater than or equal to [e'].
 
     {4 Example}
 
   {3 TypingRule.PLeq}
 
     {4 Rule}
+    All of the following applies:
+    - [p] is the pattern which matches anything lesser than or equal to an expression [e];
+    - [t_e, e'] is the result of annotating [e] in [env]; 
+    - [e'] is pure;
+    - [t] has the structure of an integer;
+    - [t_e] has the structure of an integer;
+    - [new_p] is the pattern which matches anything lesser than or equal to [e'].
 
     {4 Example}
 
   {3 TypingRule.PRange}
 
     {4 Rule}
+    All of the following applies:
+    - [p] is the pattern which matches anything within the range given by
+      expressions [e1] and [e2];
+    - [t_e1, e1'] is the result of annotating [e1] in [env];
+    - [t_e2, e2'] is the result of annotating [e2] in [env];
+    - [t] has the structure of an integer;
+    - [t_e1] has the structure of an integer;
+    - [t_e2] has the structure of an integer;
+    (*no need to check that e1' and e2' are pure here?*)
```

Comment:
Here we would need to check that `e1` and `e2` are compile-time constants..


I created HadrienRenaud/herdtools7#39 (and HadrienRenaud/herdtools7#40 which is related).